src/routes/preHandlers/courses.js

```diff
@@ -40,20 +41,21 @@ const UserCompaniesHelper = require('../../helpers/userCompanies');
 
 const { language } = translate;
 
-exports.checkAuthorization = (credentials, courseTrainerId, companies) => {
+exports.checkAuthorization = (credentials, courseTrainerId, companies, holding = null) => {
   const userVendorRole = get(credentials, 'role.vendor.name');
   const userClientRole = get(credentials, 'role.client.name');
   const userId = get(credentials, '_id');
 
   const isAdminVendor = userVendorRole === VENDOR_ADMIN;
   const isTOM = userVendorRole === TRAINING_ORGANISATION_MANAGER;
   const isTrainerAndAuthorized = userVendorRole === TRAINER && UtilsHelper.areObjectIdsEquals(userId, courseTrainerId);
-  const isClientOrHoldingAndAuthorized = [CLIENT_ADMIN, COACH].includes(userClientRole) &&
-    companies.some(company => UtilsHelper.hasUserAccessToCompany(credentials, company));
 
-  if (!isAdminVendor && !isTOM && !isTrainerAndAuthorized && !isClientOrHoldingAndAuthorized) {
-    throw Boom.forbidden();
-  }
+  const hasAccessToHolding = UtilsHelper.areObjectIdsEquals(holding, get(credentials, 'holding._id'));
+  const hasAccessToCompany = companies.some(company => UtilsHelper.hasUserAccessToCompany(credentials, company));
+  const isClientAndAuthorized = [CLIENT_ADMIN, COACH].includes(userClientRole) &&
+    (hasAccessToHolding || hasAccessToCompany);
```

Comment:
est ce qu'ici ce ne serait pas plutot
```
const isClientAndAuthorized = [CLIENT_ADMIN, COACH].includes(userClientRole) && hasAccessToCompany;
const isHoldingAndAuthorized = userholdingRole === HOLDING_ADMIN && hasAccessToHolding;
```

Reply:
je trouve ca bizarre de considerer la holding sans prendre en compte son role HOLDING_ADMIN
et aussi on s'etait noté ca au moment de la conception du MVP
![image](https://github.com/sophiemoustard/compani-api/assets/24521863/6b691416-d38f-4461-84b1-6de71c259b80)

Reply:
si il y a une holding dans ces credentials, il a forcément un rôle holding_admin + il est doit être client_admin/coach donc en soit non c'est pas une erreur, mais oui je peux l'écrire de ta façon si c'est plus clair et plus carré

Reply:
par ailleurs dans ta proposition on prend aussi pour acquis qu'un holdingAdmin est client_admin ou coach car sur une formation INTRA c'est pas par isHoldingAndAuthorized qu'il accède à la formation mais par isClientAndAuthorized

Reply:
par rapport a ton premier point : justement en MVP on ne met que les coach/admin possible en admin societe mère mais a terme par forcement 
donc quand tu dis "si il y a une holding dans ces credentials, il a forcément un rôle holding_admin" je suis d'accord
par contre "+ il est doit être client_admin/coach" ca c'est le cas maintenant mais pas focrement a terme

Reply:
mais tu as raison pour ton 2eme commentaire.. et j'ai peur qu'on ait fait ca pour tous les cas en fait .... je ne l'avais pas vu sous cette angle
je comprends mieux ta premiere version de code